indigo/indigo/src/main/scala/indigo/platform/renderer/ScreenCaptureConfig.scala

```diff
@@ -0,0 +1,194 @@
+package indigo.platform.renderer
+
+import indigo.shared.ImageType
+import indigo.shared.collections.Batch
+import indigo.shared.datatypes.BindingKey
+import indigo.shared.datatypes.Rectangle
+import indigo.shared.datatypes.Vector2
+
+/** Configuration for a screen capture
+  *
+  * @param name
+  *   The optional name of the capture
+  * @param croppingRect
+  *   The rectangle to crop the capture to
+  * @param scale
+  *   The scale to apply to the capture
+  * @param excludeLayers
+  *   The layers to exclude from the capture
+  * @param imageType
+  *   The type of image to capture
+  */
+final case class ScreenCaptureConfig(
+    name: Option[String],
+    croppingRect: Option[Rectangle],
+    scale: Option[Vector2],
+    excludeLayers: Batch[BindingKey],
+    imageType: ImageType
+) {
+
+  /** Set the name of the capture
+    *
+    * @param name
+    * @return
+    */
+  def withName(name: String): ScreenCaptureConfig =
+    this.copy(name = Some(name))
```

Comment:
I noticed you doing this in the Sandbox part below and though "ah no big deal" but this is a core type so I'd better mention it:

Generally you want to avoid using the `Some(..)` constructor unless you're damn sure you're not going to get a `null` value. `Some` gives no `null` safety protection:

```scala
Some(null) // Some(null)
Option(null) // None
```

Reply:
On the other hand, using `Some` directly is marginally faster, so in performance sensitive code (which this isn't) where confidence levels are high, you might want to use `Some` instead of `Option`.